src/prefect/events/clients.py

```diff
@@ -330,14 +330,20 @@ async def _reconnect(self) -> None:
         )
 
         try:
-            message = orjson.loads(await self._websocket.recv())
+            message: dict[str, Any] = orjson.loads(await self._websocket.recv())
```

Comment:
i think we need `Dict` here right? (for 3.8)

```python
(oss-3.8) pad-2 :: github.com/PrefectHQ/prefect
» python --version
Python 3.8.17

» python -c "x: dict[str, str] = {}"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: 'type' object is not subscriptable
```

Reply:
looks like we might have also let this slip into `tests/events/test_v1_triggers_deserialization.py`

Reply:
Yikes, thank you!  I think we lost our ability to catch these in type-checking somewhere along the way.  I'll try to get that back.

Reply:
This is really weird, the tests aren't failing in 3.8 :thinking:

Reply:
hmm that is weird 🧐

Reply:
Perhaps relatedly, I was messing around with `vermin` (https://github.com/netromdk/vermin) and it found several issues in the codebase for 3.8:

```
src/prefect/client/subscriptions.py:77::!2:3.9:dict union (dict | dict)
src/prefect/events/related.py:77::!2:3.9:builtin generic type annotation (list[..])
src/prefect/server/database/configurations.py:33::!2:3.9:builtin generic type annotation (contextvars.ContextVar[..])
src/prefect/task_server.py:73::!2:3.9:builtin generic type annotation (list[..])
```